packages/react-core/src/components/Select/examples/SelectViewMore.tsx

```diff
@@ -0,0 +1,151 @@
+  const activeItemRef = React.useRef<HTMLElement>(null);
+  const viewMoreRef = React.useRef<HTMLLIElement>(null);
+  const menuRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    activeItemRef.current?.focus();
+  }, [visibleOptions]);
+
+  const simulateNetworkCall = (networkCallback: () => void) => {
+    setTimeout(networkCallback, 2000);
+  };
+
+  const getNextValidItem = (startingIndex: number, maxLength: number) => {
+    let validItem;
+    for (let i = startingIndex; i < maxLength; i++) {
+      if (selectOptions[i].props.isDisabled) {
+        continue;
+      } else {
+        validItem = selectOptions[i];
+        break;
+      }
+    }
+    return validItem;
+  };
+
+  const loadMoreOptions = () => {
+    const newLength = numOptions + 3 <= selectOptions.length ? numOptions + 3 : selectOptions.length;
+    const prevPosition = numOptions;
+    const nextValidItem = getNextValidItem(prevPosition, newLength);
+
+    setNumOptions(newLength);
+    setIsLoading(false);
+    setActiveItem(nextValidItem.props.itemId);
+    setVisibleOptions(selectOptions.slice(0, newLength));
+  };
+
+  const onViewMoreClick = () => {
+    setIsLoading(true);
+    simulateNetworkCall(() => {
+      loadMoreOptions();
+    });
+  };
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = (
+    _event: React.MouseEvent<Element, MouseEvent> | undefined,
+    itemId: string | number | undefined,
+    toggleRef: React.RefObject<any>
+  ) => {
+    // eslint-disable-next-line no-console
+    console.log('selected', itemId);
+
+    if (itemId !== 'loader') {
+      setSelected(itemId as string);
+      setIsOpen(false);
+      toggleRef?.current.focus();
+    }
+  };
+
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle
+      ref={toggleRef}
+      onClick={onToggleClick}
+      isExpanded={isOpen}
+      style={
+        {
+          width: '200px'
+        } as React.CSSProperties
+      }
+    >
+      {selected}
+    </MenuToggle>
+  );
+
+  return (
+    <Select
+      id="single-view-more-select"
+      ref={menuRef}
+      isOpen={isOpen}
+      selected={selected}
+      onSelect={onSelect}
+      onOpenChange={(isOpen) => setIsOpen(isOpen)}
+      toggle={toggle}
+    >
+      <SelectList>
+        {visibleOptions.map((option) => {
+          const props = option.props;
+
+          return <SelectOption key={option.key} ref={props.itemId === activeItem ? activeItemRef : null} {...props} />;
+        })}
+        {numOptions !== selectOptions.length && (
+          <SelectOption
+            {...(!isLoading && { isLoadButton: true })}
+            {...(isLoading && { isLoading: true })}
+            onClick={onViewMoreClick}
+            itemId="loader"
+            ref={viewMoreRef}
+          >
+            {isLoading ? <Spinner size="lg" /> : 'View more'}
+          </SelectOption>
+        )}
+      </SelectList>
```

Comment:
This is probably fine. Only possible issue that I wouldn't block over would be that this gets announced as though it were an option to select. The text content would probably provide enough context. An alternative might be using a Button outside of the SelectList, though that would require settingup additional keyboard handling in the example.

Reply:
Is the announcement due to the `role="menuitem"` that `MenuItem` places on its element? We may be able to get around the announcing without having to re-wire keyboard handling by exposing `role` as a prop and passing in a role manually with the view more option. Do you think changing the role would work?

Reply:
Removing/changing the role would flag an error in Axe. Since the Select is using a listbox role, it has to contain children with the "option" role.

I'd be fine with this as-is and revisit it. One possible alternative would be doing something similar to what you did in #9014 and add the view more button to a footer.

Reply:
Adding the button to the footer would change the keyboard behavior a decent amount is the drawback, because the footer is navigated to via `tab` and wouldn't focus the next item in the list.